### PR TITLE
Android Issue #91: Added confirmation popup for cancel button.

### PR DESCRIPTION
**Pull request by Shradhajoshi**
TODO: Link the issue #91 

TODO: Request a maintainer to review your PR
* @baghaii , @gaurigadkari , @renweview  Kindly review my PR

TODO: Write a brief description of what your PR accomplished
* Added confirmation popup for cancel.

## TODO: Explain how to test your PR
* Go to TaskDetailScreen->Press on cancel button

### CODE REVIEW

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt [hunk 1]
```diff
@@ -181,9 +192,57 @@ class TaskDetailScreen {
...
+                    onClick = { onConfirm() }
+                ) {
+                    Text(text = "Ok")
+                }
+            },
+            dismissButton = {
+                Button(
+                    onClick = { onCancel() }
+                ) {
+                    Text(text = "Cancel")
```
Comment (baghaii) [inline coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:226]: Can you use the `stringResource(id = R.string.cancel)` here?

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt [hunk 2]
```diff
@@ -181,9 +192,57 @@ class TaskDetailScreen {
...
+        onCancel: () -> Unit
+    ) {
+        AlertDialog(
+            onDismissRequest = onCancel,
+            text = { Text(text = message) },
+            confirmButton = {
+                Button(
+                    onClick = { onConfirm() }
+                ) {
+                    Text(text = "Ok")
```
Comment (baghaii) [inline coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:219]: Can you move the "OK" to the string resources file and use a stringResource here?

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt [hunk 3]
```diff
@@ -49,7 +50,15 @@ import com.example.tasktracker.ui.theme.Green
 class TaskDetailScreen {
     @OptIn(ExperimentalMaterial3Api::class)
     @Composable
-    fun TaskDetailScreen() {
+    fun TaskDetailScreen(onNavigateToList: () -> Unit) {
```
Comment (baghaii) [inline coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:53]: I like that you added this piece preparing us for navigation.

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt (hunk 2 above)
Comment (Shradhajoshi) [reply coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt]: Addressed.

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt (hunk 3 above)
Comment (Shradhajoshi) [reply coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:53]: Addressed.

**Review by renweview — APPROVED** [review]
Clean, stuck with the protocol @Shradhajoshi good job and thank you!

**Pull request merged**